### PR TITLE
fix(splitter) Abandon column splitter if the resulting query may be too large.

### PR DESCRIPTION
**Pull request by fpacifici**
We have reports of queries with high result limit (10000) that break when running the column splitter because the list of event ids in the generated query is so long that the resulting query beyond clickhouse limit.

So this is the quickest fix to make column splitter more conservative and abandon if the initial limit is too high or if the generated query is too long.

### CODE REVIEW

File: snuba/web/split.py [hunk 1]
```diff
@@ -384,6 +387,11 @@ def execute(
         event_ids = list(
             set([event[self.__id_column] for event in result.result["data"]])
         )
+        if len(event_ids) > settings.COLUMN_SPLIT_MAX_RESULTS:
+            # We may be runing a query that is beyond clickhouse maximum query size,
+            # so we cowardly abandon.
+            return None
```
Comment (tkaemming) [inline snuba/web/split.py:393]: Should this be tracked so we know how frequently it's occurring? This seems detrimental if we're doing it frequently.

File: snuba/web/split.py [hunk 2]
```diff
@@ -384,6 +387,11 @@ def execute(
         event_ids = list(
             set([event[self.__id_column] for event in result.result["data"]])
         )
+        if len(event_ids) > settings.COLUMN_SPLIT_MAX_RESULTS:
+            # We may be runing a query that is beyond clickhouse maximum query size,
+            # so we cowardly abandon.
```
Comment (tkaemming) [inline snuba/web/split.py:392]: lol

File: snuba/settings.py [hunk 3]
```diff
@@ -116,6 +116,9 @@
     },
 }  # (dataset name: (referrer: percentage))
 
+COLUMN_SPLIT_MAX_LIMIT = 1000
+COLUMN_SPLIT_MAX_RESULTS = 5000
```
Comment (tkaemming) [inline snuba/settings.py:120]: Any basis for these numbers or are they just approximations?

**Review by tkaemming — APPROVED** [review]

File: snuba/settings.py (hunk 3 above)
Comment (fpacifici) [reply snuba/settings.py:120]: The limit is the default query limit, which is the maximum sentry uses (other than csv exporter which is causing the issue).
Max result is 32 bytes (the string event it converted into bytes we send to clickhouse) times 5000 ->  160kb. I don't know if clickhouse measures that based on 8 bit characters or unicode, so I gave it twice as much. + Margin for the rest of the query to be comfortably under the 500kb limit.

Will check how clickhouse measures later.

File: snuba/web/split.py (hunk 1 above)
Comment (fpacifici) [reply snuba/web/split.py:393]: makes sense

**Pull request merged**